### ISSUE TITLE
Dependency on ifconfig

### ISSUE THREAD

**Issue by bannsec** [GitSquared/edex-ui#227]
Looks like there's an implicit dependency on ifconfig being present. Probably a good idea to have a graceful fallback on that, as in my case it wasn't installed by default, and in some distros (such as Ubuntu), it's being moved away from completely in preference to `ip`. Perhaps check for both and fail gracefully if neither are found.

**Comment by GitSquared** [comment]
This is probably due to [systeminformation](https://github.com/sebhildebrandt/systeminformation). I'll investigate this more deeply.

Can you confirm not having `ifconfig` installed does prevent some features to work correctly / gives you an error message?

**Comment by bannsec** [comment]
I can confirm not having it installed causes errors. However, there seem to be a few different issues. Installing net-tools did stop that error from coming up on the underlying console, but i'm not sure what effect it had on the ui.

**Comment by GitSquared** [comment]
Could you give me the exact error? I think i know where it happens but i want to be sure.

**Comment by bannsec** [comment]
Not an error that your console mentions, it's literally bash outputting this error over and over:

```
/bin/sh: 1: /sbin/ifconfig: not found
```

**Comment by eadwu** [comment]
On NixOS, it doesn't seem to matter if `ifconfig` is on the $PATH.
```
/bin/sh: /sbin/ifconfig: No such file or directory
✖  error     TypeError: Cannot read property 'replace' of undefined: Uncaught TypeError: Cannot read property 'replace' of undefined
⬤  debug     at $HOME/.cache/appimage-run/6e2f4142bf0a212a945d51ea7e53277098e5894b32cc66d0438a76ffc8425f3b/squashfs-root/resources/app.asar/node_modules/systeminformation/lib/memory.js 120:32
```

**Comment by GitSquared** [comment]
So, funny thing, I actually [reported this](https://github.com/sebhildebrandt/systeminformation/issues/114) on systeminformation a while back. I [proposed a solution](https://github.com/sebhildebrandt/systeminformation/pull/115) that the maintainer of this project used to fix the issue.

tl;dr of the whole conversation is that he added a test to see if `ip` is available and the library would either use that or fallback to `ifconfig`. This was in March 2018 and that code is still up

I double-checked the possibly [related](https://github.com/GitSquared/edex-ui/blob/master/src/classes/netstat.class.js) [files](https://github.com/GitSquared/edex-ui/blob/master/src/classes/conninfo.class.js) in eDEX, but i just don't see where this `ifconfig` launch is coming from.

Just thought i'd share some progress.

**Comment by GitSquared** [comment]
OK I think i got it. [Look at how systeminformation checks for `ip`](https://github.com/sebhildebrandt/systeminformation/blob/e66f0c984492d13ca211de8fd03d00896fc70f1e/lib/network.js#L75).

Could it be that the `ip` exec is elsewhere than `/sbin/` ? @bannsec @eadwu 
If so i'd like to know where it's stored on your machines and i'll send a PR up to the library.

**Comment by eadwu** [comment]
Yes, /sbin doesn't exist in NixOS, but they exist on the $PATH, so using `which` or `command -v` would be better, though there is a symlink at `/run/current-system/sw/bin/ip`.

**Comment by GitSquared** [comment]
Thanks for the confirmation. I'll make an upstream PR.

**Comment by bannsec** [comment]
Also ip and ifconfig are both technically in optional packages and so may
not be present.

**Comment by GitSquared** [comment]
@eadwu @bannsec Apart from the edgecase where neither ip nor ifconfig are present, can you confirm the issue is resolved?

**Comment by eadwu** [comment]
Looking good so far. https://i.imgur.com/qgaHztl.jpg

**Comment by bannsec** [comment]
Honestly loosing track of which errors are associated with which tickets. In any case, here's my current image. It does claim i'm offline, and no network usage tracking graph.
![current](https://user-images.githubusercontent.com/7632864/48957747-432fd400-ef20-11e8-905d-54cbc76e07a0.png)

**Comment by GitSquared** [comment]
Should be good regarding to this issue i think. Closing.
